File: src/main/kotlin/easy/array/MatrixDiagonalSum.kt
package easy.array

fun diagonalSum(mat: Array<IntArray>): Int {
    var sum = 0

    val size = mat.size
    for (i in mat.indices) {
        for (j in mat[i].indices) {
            when {
                i == j || j == size - 1 - i -> sum += mat[i][j]
            }
        }
    }
    return sum
}

File: src/main/go/easy/array/matrix_diagonal_sum.go
package array

func diagonalSum(mat [][]int) int {
    sum := 0

    size := len(mat)
    for i := range size {
        for j := range size {
            switch {
            case i == j || j == size-1-i:
                sum += mat[i][j]
            }
        }
    }
    return sum
}
